networks/suzuka/suzuka-client/src/tests/settlement_state.rs

```diff
@@ -0,0 +1,217 @@
+	faucet_client.fund(bob.address(), 100_000_000).await?;
+	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
+	// Have Alice send Bob some coins.
+	let txn_hash = coin_client.transfer(&mut alice, bob.address(), 1_000, None).await?;
+	rest_client.wait_for_transaction(&txn_hash).await?;
+	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(2000)).await;
+
+	// Read Alice and bod account balance that shouldn't be updated on fin state.
+	let finwiew_aptos_client = AptosClient::new(finview_node_url.clone().parse()?);
+	let fin_view_coin_client = CoinClient::new(&finwiew_aptos_client);
+	// It should be in error because the account shouldn't be funded.
+	let before_finwiew_alice_balance =
+		fin_view_coin_client.get_account_balance(&alice.address()).await;
+	assert!(before_finwiew_alice_balance.is_err());
+	let before_finwiew_bob_balance = fin_view_coin_client.get_account_balance(&bob.address()).await;
+	assert!(before_finwiew_bob_balance.is_err());
+
+	let final_alice_balance = coin_client.get_account_balance(&alice.address()).await?;
+	let final_bob_balance = coin_client.get_account_balance(&bob.address()).await?;
+
+	// The node should have produced one or several blocks with Alice and Bod Tx.
+	// Make these blocks commitment accepted. We suppose there's less than 10 blocks.
+	let mut last_seen_height = 0;
+	for index in 1..10 {
+		// Get the associated commitment.
+		let MCR::getValidatorCommitmentAtBlockHeightReturn { _0: onchain_commitment } =
+			validator_mcr
+				.getValidatorCommitmentAtBlockHeight(
+					last_seen_commitment + U256::from(index),
+					validator_private_key.address(),
+				)
+				.call()
+				.await?;
+		if onchain_commitment.height != U256::from(0) {
+			last_seen_height = onchain_commitment.height.try_into()?;
+			let commitment = BlockCommitment {
+				height: last_seen_height,
+				block_id: Id(onchain_commitment.blockId.into()),
+				commitment: Commitment(onchain_commitment.commitment.into()),
+			};
+			validator2_client.post_block_commitment(commitment).await?;
+		} else {
+			break;
+		}
+	}
+
+	// Wait the accepted commitment to be finalized. Can take more than 10s for all finality state.
+	let _ = tokio::time::sleep(tokio::time::Duration::from_millis(20000)).await;
+	let after_finview_alice_balance = fin_view_coin_client
+		.get_account_balance(&alice.address())
+		.await
+		.context("Failed to get final Alice's account balance")?;
+	let after_finview_bob_balance = fin_view_coin_client
+		.get_account_balance(&bob.address())
+		.await
+		.context("Failed to get final Bob's account balance")?;
+
+	//Alice and bob balance should have been finalized in finview
+	assert_eq!(after_finview_alice_balance, final_alice_balance);
+	assert_eq!(after_finview_bob_balance, final_bob_balance);
+
+	// verify node finality state with block stored at the commitment height.
+	let response = restclient.get(&fin_state_root_hash_url).send().await?;
```

Comment:
Answered on Slack: if we want to calculate the commitments as part of the checks done by test, this is not enough, ledger info is needed.

It's a matter of testing philosophy whether we want to actually verify that the commitment hash is as expected. This may be too much to ask from an e2e test and is better verified by protocol unit tests.

Reply:
I remove the fin state verification